### PR TITLE
Added Core.hs to each era that has a TxBody class

### LINKED ISSUE #2932: Classes defined in each era should go into `Core` module
It was discussed that classes are spread around through out the repo and it would be best if each era defined type classes in the `Cardano.Ledger.[Era].Core` module. Unless of course the type class is used as an internal implementation in some module.

For example newly defined type classes in #2901, eg `AlonzoEraTxBody` is defined in `Cardano.Ledger.Alonzo.TxBody` module, but it needs to move into `Cardano.Ledger.Alonzo.Core`, while instances should still live in `Cardano.Ledger.Alonzo.TxBody`.

For convenience I also suggest we re-export the whole of `Cardno.Ledger.Core` module and `Cardano.Ledger.[Era].Core` from a previous era from it as well. Eg.

```haskell
module Cardano.Ledger.Shelley.Core
  ( module Cardano.Ledger.Core
  , ShelleyTxBody(..)
  ...
  ) where
```

```haskell
module Cardano.Ledger.ShelleyMA.Core
  ( module Cardano.Ledger.Shelley.Core
  , ShelleyMATxBody(..)
  ...
  ) where
```

### PR DESCRIPTION
**Pull request by Soupstraw**
This PR moves the `[Era]EraTxBody` classes from each era to the `Core` module of that era.

closes #2932

### CODE REVIEW

File: eras/conway/impl/src/Cardano/Ledger/Conway/Core.hs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module Cardano.Ledger.Conway.Core
```
Comment (teodanciu) [inline eras/conway/impl/src/Cardano/Ledger/Conway/Core.hs]: Not sure about this, does it bring any value at this point, or better to add it when needed? I don't necessarily mind, just wondering

**Review by teodanciu — COMMENTED** [review]
Looks good to me. 
I noticed we didn't do the suggestion from the ticket to 
> re-export the whole of Cardno.Ledger.Core module and Cardano.Ledger.[Era].Core from a previous era from it as well

- by choice or an omission?

File: eras/conway/impl/src/Cardano/Ledger/Conway/Core.hs (hunk 1 above)
Comment (Soupstraw) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Core.hs]: I actually didn't mean to push this file

**Comment by Soupstraw** [comment]
> Looks good to me. I noticed we didn't do the suggestion from the ticket to
> 
> > re-export the whole of Cardno.Ledger.Core module and Cardano.Ledger.[Era].Core from a previous era from it as well
> 
>     * by choice or an omission?

Fixed it now

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/Core.hs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  ( ShelleyEraTxBody (..),
+    Wdrl (..),
+    module Cardano.Ledger.Core,
+  )
+where
+
+import Cardano.Binary (FromCBOR (..), ToCBOR (..))
+import Cardano.Ledger.Address (RewardAcnt (..))
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Core (Era (..), EraTxBody (..), ExactEra)
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/Core.hs]: The idea was to re-export the whole of `Core`, so in all other modules we can import `import Cardano.Ledger.Shelley.Core` and not bother with `import Cardano.Ledger.Core`. For this we need to import that module as a whole
```suggestion
import Cardano.Ledger.Core
```

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Core.hs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+module Cardano.Ledger.ShelleyMA.Core
+  ( ShelleyMAEraTxBody (..),
+    module Cardano.Ledger.Shelley.Core,
+  )
+where
+
+import Cardano.Ledger.Core (Era (..), EraTxBody (..), Value)
+import Cardano.Ledger.Mary.Value (MultiAsset (..))
+import Cardano.Ledger.Shelley.Core (ShelleyEraTxBody)
```
Comment (lehins) [inline eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Core.hs]: So, here we no longer need to import `Cardano.Ledger.Core`, but we re-export the whole of `Cardano.Ledger.Shelley.Core`:
```suggestion
```
```suggestion
import Cardano.Ledger.Mary.Value (MultiAsset (..))
import Cardano.Ledger.Shelley.Core
```

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/Core.hs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+module Cardano.Ledger.Babbage.Core
+  ( BabbageEraTxBody (..),
+    module Cardano.Ledger.Alonzo.Core,
+  )
+where
+
+import qualified Cardano.Ledger.Alonzo.Core as X
+import Cardano.Ledger.Babbage.TxOut (BabbageEraTxOut)
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Core (Era (..), EraTxBody (..), EraTxOut (..))
```
Comment (lehins) [inline eras/babbage/impl/src/Cardano/Ledger/Babbage/Core.hs]: Same here
```suggestion
import Cardano.Ledger.Alonzo.Core
import Cardano.Ledger.Babbage.TxOut (BabbageEraTxOut)
import Cardano.Ledger.Coin (Coin (..))
```

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Core.hs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    module Cardano.Ledger.ShelleyMA.Core,
+  )
+where
+
+import Cardano.Ledger.Alonzo.TxOut (AlonzoEraTxOut)
+import Cardano.Ledger.BaseTypes (Network)
+import Cardano.Ledger.Core (Era (..), EraIndependentScriptIntegrity, EraTxBody (..))
+import Cardano.Ledger.Keys (KeyHash (..), KeyRole (..))
+import Cardano.Ledger.SafeHash (SafeHash)
+import Cardano.Ledger.ShelleyMA.Core (ShelleyMAEraTxBody)
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Core.hs]: Same here
```suggestion
import Cardano.Ledger.Keys (KeyHash (..), KeyRole (..))
import Cardano.Ledger.SafeHash (SafeHash)
import Cardano.Ledger.ShelleyMA.Core
```

**Review by lehins — CHANGES_REQUESTED** [review]
Just one conceptual change need, otherwise looks great.

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/Core.hs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/Core.hs]: No orphans here
```suggestion
```

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/Core.hs [hunk 7]
```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/Core.hs]: Unused extensions
```suggestion
```

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/Core.hs [hunk 8]
```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE DataKinds #-}
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/Core.hs]: Unused extension
```suggestion
```

**Review by lehins — APPROVED** [review]
A few redundant things, but otherwise looks great

**Pull request merged**